test/OpenTelemetry.Instrumentation.AspNetCore.Tests/MetricTests.cs

```diff
@@ -84,32 +84,33 @@ public async Task ValidateNet8MetricsAsync()
         var res = await client.GetStringAsync("http://localhost:5000/").ConfigureAwait(false);
         Assert.NotNull(res);
 
+        // We need to let metric callback execute as it is executed AFTER response was returned.
+        // In unit tests environment there may be a lot of parallel unit tests executed, so
+        // giving some breezing room for the callbacks to complete
+        await Task.Delay(TimeSpan.FromSeconds(1)).ConfigureAwait(false);
```

Comment:
This seems super unreliable. Why not use a mock listener ?

Reply:
what is a mock listener here? MeterListener?

Reply:
@vishweshbankwar If you performed `await app.DisposeAsync();` here would that cause everything to flush out in a more deterministic way?

Reply:
You could keep it how you have it. Just dispose both...

```csharp
await app.DisposeAsync(); // Flush AspNetCore
this.meterProvider.Dispose(); // Flush manual provider
```

Should be fine.

You could do it all from the host if you wanted to like this...

```csharp
        var metricItems = new List<Metric>();

        var builder = WebApplication.CreateBuilder();
        builder.Logging.ClearProviders();

        builder.Services.AddOpenTelemetry()
            .WithMetrics(metrics => metrics
                 .AddAspNetCoreInstrumentation()
                 .AddInMemoryExporter(metricItems));

        var app = builder.Build();

        app.MapGet("/", () => "Hello");

        _ = app.RunAsync();

        using var client = new HttpClient();
        var res = await client.GetStringAsync("http://localhost:5000/").ConfigureAwait(false);
        Assert.NotNull(res);

       await app.DisposeAsync(); // This should flush AspNetCore and the MeterProvider in this case

     // inspect metricItems here it should be populated with everything that happened
```

Reply:
Thanks! - I will follow up on this. Opened https://github.com/open-telemetry/opentelemetry-dotnet/issues/4972 for tracking